### PR TITLE
Translations/ add helper function for OpenAI prompt text generator

### PR DESCRIPTION
**Pull request by eadams17**
## WHAT
add helper function for OpenAI prompt text generator to be more descriptive for certain locales

## WHY
some of the locales are badly translated by OpenAI due to the prompt not being descriptive enough-- this update provides the full language name as well as the script type for that language

## HOW
add a helper function that conditionally renders more descriptive text for certain locales

### Screenshots
(If applicable. Also, please censor any sensitive data)
Screen Shot 2024-11-27 at 12 27 41 PM

### Notion Card Links
(Please provide links to any relevant Notion card(s) relevant to this PR.)
https://www.notion.so/quill/Translation-Project-Complete-Translations-of-other-Languages-11dd42e6f94180559387d370d57ea7af?pvs=4

### What have you done to QA this feature?
(Provide enough detail that a reviewer could assess whether additional QA should be done. For larger projects, additionally use the Engineer Feature Testing Notion template. Review Guidelines if needed: https://www.notion.so/quill/Github-PR-QA-Guidelines-49e99fc965654ceeb8c6249bd9d181d7)
verified on staging that more accurate translations are supplied

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  yes
Have you deployed to Staging? | yes
Self-Review: Have you done an initial self-review of the code below on Github? | yes

### CODE REVIEW

File: services/QuillLMS/app/models/translation_prompts.rb [hunk 1]
```diff
@@ -22,4 +22,15 @@ def self.feedback_strings_custom_prompt
   def self.feedback_strings_prompt(locale:)
     prompt_start(locale:) + feedback_strings_custom_prompt
   end
+
+  def self.format_locale(locale)
```
Comment (brendanshean) [inline services/QuillLMS/app/models/translation_prompts.rb]: Let's use a hash stored as a constant, which saves us memory from not having to do string allocations each time the method is called

```
  LOCALE_SCRIPTS = {
    'hi' => 'Hindi (Devangari script)',
    'ur' => 'Urdu (Nastaliq script)',
    'ko' => 'Korean (Hangul script)',
    'uk' => 'Ukrainian (Cyrillic script)',
    'ts' => 'Tagalog (Latin script)',
    'prs' => 'Dari (Perso-Arabic script)'
  }.freeze
  
  def self.format_locale(locale) = LOCALE_SCRIPTS.fetch(locale, locale)
```

File: services/QuillLMS/spec/models/translation_prompts_spec.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe TranslationPrompts, type: :model do
+  describe '.prompt_start' do
```
Comment (brendanshean) [inline services/QuillLMS/spec/models/translation_prompts_spec.rb:6]: Here's some refactoring to clean up the duplication:

```
describe '.prompt_start' do
  subject { TranslationPrompts.prompt_start(locale:) }

  context 'hindi' do
    let(:locale) { 'hi' }

    it { is_expected.to include('You are going to do a translation from english to Hindi (Devangari script)') }
  end

  context 'korean' do
    let(:locale) { 'ko' }

    it { is_expected.to include('You are going to do a translation from english to Korean (Hangul script)') }
  end

  context 'not in the predefined list' do
    let(:locale) { 'fr' }

    it { is_expected.to include('You are going to do a translation from english to fr') }
  end
end
```

File: services/QuillLMS/spec/models/translation_prompts_spec.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  describe '.feedback_strings_prompt' do
+    it 'combines the start prompt and JSON prompt' do
+      locale = 'uk'
+      result = TranslationPrompts.feedback_strings_prompt(locale: locale)
+      expect(result).to include('You are going to do a translation from english to Ukrainian (Cyrillic script)')
+      expect(result).to include('Following is the text to translate in a JSON format.')
+    end
+  end
+
+  describe '.format_locale' do
```
Comment (brendanshean) [inline services/QuillLMS/spec/models/translation_prompts_spec.rb:45]: Some more refactoring:


```
describe '.format_locale' do
  subject { TranslationPrompts.format_locale(locale) }

  context 'hindi' do
    let(:locale) { 'hi' }

    it { is_expected.to eq('Hindi (Devangari script)') }
  end

  context 'urdu' do
    let(:locale) { 'ur' }

    it { is_expected.to eq('Urdu (Nastaliq script)') }
  end

  context 'not in the predefined list' do
    let(:locale) { 'es' }

    it { is_expected.to eq('es') }
  end
end
```

**Review by brendanshean — APPROVED** [review]
I added some refactoring comments, but otherwise looks good to go after that.

**Comment by eadams17** [comment]
> I added some refactoring comments, but otherwise looks good to go after that.

thank you for these suggestions! these are great, I'll make these updates

**Pull request merged**